### DISCUSSION TITLE
Unable to close overlay by clicking on trigger a second time

### DISCUSSION THREAD

**Discussion by areknow** [adobe/react-spectrum#1820] [Q&A] [answered]
Hello, thank you for the library 🙏

I would like assistance in solving this issue. I have a button trigger that opens a overlay using useOverlay and useOverlayPosition. When I click the button a second time, nothing happens. I would like the be able to close the overlay with a second click of the trigger AND by clicking off the overlay content.
https://codesandbox.io/s/react-aria-useoverlay-useoverlayposition-hmfmc?file=/src/App.js

Thanks

**Comment by snowystinger** [comment, marked as answer]
Hey, I just took a look, I think you are just missing useOverlayTrigger as well, see our example here
https://react-spectrum.adobe.com/react-aria/useOverlayTrigger.html#example

**Reply by areknow** [reply]
Thanks for the reply! In the example you posted, if you click on the button after opening the dialog, the dialog does not close. 
None the less, I did try adding the useOverlayTrigger but there is no change. 
I updated the sandbox: https://codesandbox.io/s/react-aria-useoverlay-useoverlayposition-hmfmc?file=/src/App.js

**Reply by snowystinger** [reply]
Ah yep, looks like we have just `state.open()` in there, it should be `state.toggle()`

**Reply by snowystinger** [reply]
https://codesandbox.io/s/react-aria-useoverlay-useoverlayposition-forked-qe7vl?file=/src/App.js

**Reply by areknow** [reply]
Duh that makes perfect sense! Thank you very much!
